crates/codegen/runtime/cargo/crate/src/runtime/parser/parse_output.rs

```diff
@@ -1,15 +1,17 @@
-use crate::cst::{Cursor, Node, TextIndex};
+use std::rc::Rc;
+
+use crate::cst::{Cursor, NonterminalNode, TextIndex};
 use crate::parser::ParseError;
 
 #[derive(Debug, PartialEq)]
 pub struct ParseOutput {
-    pub(crate) parse_tree: Node,
+    pub(crate) parse_tree: Rc<NonterminalNode>,
     pub(crate) errors: Vec<ParseError>,
 }
 
 impl ParseOutput {
-    pub fn tree(&self) -> Node {
-        self.parse_tree.clone()
+    pub fn tree(&self) -> Rc<NonterminalNode> {
```

Comment:
TOL: I wonder why cloning by default? WDYT of returning `&Rc<>` and leave cloning up to the user? This is similar to other APIs like `Node::as_nonterminal()`.

I realize that `Rc::clone()` is verbose, but we decided to enable a lint rule that enforces that for clarity (we can revisit this decision separately).